presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -119,33 +119,13 @@ public Expression rewriteExpression(Expression node, Void context, ExpressionTre
                 if (expressionToSymbols.containsKey(node)) {
                     return expressionToSymbols.get(node).toSymbolReference();
                 }
-                else if (expressionToExpressions.containsKey(node)) {
-                    Expression mapping = getMapping(node);
-                    mapping = translateNamesToSymbols(mapping);
-                    return treeRewriter.defaultRewrite(mapping, context);
-                }
-                else {
-                    return treeRewriter.defaultRewrite(node, context);
-                }
+
+                Expression translated = expressionToExpressions.getOrDefault(node, node);
+                return treeRewriter.defaultRewrite(translated, context);
             }
         }, mapped);
     }
 
-    private Expression getMapping(Expression expression)
-    {
-        if (!expressionToExpressions.containsKey(expression)) {
-            return expression;
-        }
-
-        Expression mapped = expressionToExpressions.get(expression);
-        Expression translated = translateNamesToSymbols(mapped);
```

Comment:
Maybe `translateNamesToSymbols` is required after each `expressionToExpressions`? 
Could `expressionToExpressions` contain untranslated symbols?

Reply:
That shouldn't be the case. The only place that modifies this map does:

```
expressionToExpressions.put(translateNamesToSymbols(expression), rewritten)
```

But it's possible the caller is passing the wrong expression for "rewritten".

It may be the case that this change only works properly once addIntermediateMappings is removed (in the other PR). If so, I'll squash them together.

Reply:
Ah, yes, it's because of how planning of quantified comparisons works vs intermediate mappings. It's (incorrectly) relying on the TranslationMap that to do a syntactic rewrite. This gets fixed by the change to remove intermediate mappings.

Reply:
Why not to change:
```
expressionToExpressions.put(translateNamesToSymbols(expression), rewritten)
```
to
```
expressionToExpressions.put(translateNamesToSymbols(expression), translateNamesToSymbols(rewritten))
```
?

Alternatively you could add check state that `rewritten` doesn't contain any `Identifiers`

Reply:
After this PR, the only usages of `put(Expression, Expression)` are in SubqueryPlanner, so that would seem reasonable. I'll look into it later. It may be possible to get rid of the method altogether.